file/src/main/scala/akka/stream/alpakka/file/javadsl/LogRotatorSink.scala

```diff
@@ -5,43 +5,73 @@
 package akka.stream.alpakka.file.javadsl
 
 import java.nio.file.{Path, StandardOpenOption}
-import java.util
 import java.util.Optional
 import java.util.concurrent.CompletionStage
 
 import akka.Done
 import akka.stream.javadsl
+import akka.stream.scaladsl
 import akka.stream.javadsl.Sink
 import akka.util.ByteString
-import akka.japi.{function}
+import akka.japi.function
 
 import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+import scala.compat.java8.FutureConverters._
+import scala.compat.java8.OptionConverters._
 
 object LogRotatorSink {
+
+  /**
+   * Sink directing the incoming `ByteString`s to new files whenever `generatorFunction` returns a new value.
```

Comment:
There is a mismatch of how the argument is named in the declaration of the method and the comment.